### PR TITLE
Improve `Embedded` rule

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -468,22 +468,22 @@ Common flow is the same as you would use usual classes:
...
-        new Embedded(Point::class),
+        new Embedded(),
     ])]
     private array $points;
 }
 
 final class Point
 {
-    #[Embedded(Coordinates::class)]
+    #[Embedded]
```
Comment (xepozz) [inline README.md:486]: It should also work with specifying a class because the value may be a scalar value.

**Review by xepozz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (vjik) [reply README.md:486]: `Embedded` rule for scalar value? Why? 
Can you describe example more detailed?

File: src/Rule/Trait/BeforeValidationTrait.php [hunk 2]
```diff
@@ -19,6 +19,9 @@ public function getSkipOnEmpty(): bool
         return $this->skipOnEmpty;
     }
 
+    /**
+     * @psalm-suppress InvalidNullableReturnType NEED FIX BEFORE RELEASE!!!
```
Comment (samdark) [inline src/Rule/Trait/BeforeValidationTrait.php:23]: ?

**Review by samdark — APPROVED** [review]

File: src/Rule/Trait/BeforeValidationTrait.php (hunk 2 above)
Comment (vjik) [reply src/Rule/Trait/BeforeValidationTrait.php:23]: Fix this psalm error required modify many files. I'll make this in separate PR.

**Comment by vjik** [comment]
Closed in favor #297

**Pull request closed without merging**